### ISSUE TITLE
Program stuck on "Gathering installed games..."

### ISSUE THREAD

**Issue by YuiKitamura** [Drommedhar/DlssUpdater#9]
After updating to version 1.0.5.1, the program sits at gathering installed games forever.

My installed games are different from the first time I ran the program, not sure if that caused this to happen or if it was the update.

**Comment by Drommedhar** [comment]
Could you please attach your logfiles?
You can find them in the "log" directory next to the executable.

**Comment by YuiKitamura** [comment]
[AppLog.2024-10-02.txt](https://github.com/user-attachments/files/17220797/AppLog.2024-10-02.txt)

**Comment by Drommedhar** [comment]
I might know whats causing this.
Will get to it later today after work and push an update. I hope this will help. We will see,

**Comment by Drommedhar** [comment]
I just pushed update 1.0.6.0 which hopefully fixes your issue.
Could you please check this with this version?

If it still does hang on startup, try renaming the "settings" folder so it starts with a fresh set. If it works after that, I might have a bug somewhere else.

**Comment by YuiKitamura** [comment]
It still got stuck on the 1.0.6.0 update. Deleting the settings folder and running the program again did result in the program opening.

**Comment by Drommedhar** [comment]
Has the issue occured again with 2.0?

**Comment by Drommedhar** [comment]
It looks like in your 2 library files there are 531 game entries.
It will try to look for each of it, which will take some time.

And none of the games is checked a second time which it would periodically do. So it really is still checking your steam library. 
What I will do is trying to speed up the gathering step, so more games are checked in parallel. Can I somehow get a test version to you (like through discord)?

**Comment by YuiKitamura** [comment]
Feel free to DM me @yuikitamura on Discord

**Comment by YuiKitamura** [comment]
Updated to version 2.0.4.0. It still takes quite a while for it to boot into the program the first time/after an update. Somewhere around 5 minutes. Subsequent boots only take about 15 seconds. I've additionally noticed that about half of my games are listed twice for some reason.

**Comment by Drommedhar** [comment]
Thats even more strange that subsequent boots are faster. It always queries each library on start. 
Also games should not appear twice as they are filtered by path. So the question would be are these games from an automatically detected library or manually added games?
Do you have a log file from the new version? I increased logging output slightly, maybe I will see whats going on.

**Comment by YuiKitamura** [comment]
[AppLog.2024-10-10.txt](https://github.com/user-attachments/files/17351540/AppLog.2024-10-10.txt)

**Comment by Drommedhar** [comment]
The newest version "2.0.5.2" (or even the one before) should fix the games appearing doubled.
Has anything changed regarding startup?

**Comment by YuiKitamura** [comment]
So I've wiped my computer for unrelated reasons and just downloaded the 2.0.5.2 version and was given a crash error. Let me know if you would prefer I open a separate issue for this.

I tried to upload the dmp file but there's a limit of 25MB and the file size is 436MB

**Comment by maicol07** [comment]
Hi, I have the same issue, I've just uploaded the dump on an external file sharing website: https://file.io/n56IgdcyaNPd

**Comment by maicol07** [comment]
@Drommedhar it seems the issue occurs when `GamePath` or `GameImgeUri` don't exist anymore.

**Comment by Drommedhar** [comment]
@maicol07 Thanks for the information. I'm currently pretty busy with work, but hope to get to fixing stuff maybe at the end of year in my vacation.
